### DISCUSSION TITLE
Out of the box https solution for local development

### DISCUSSION THREAD

**Discussion by ericfennis** [vercel/next.js#33768] [Ideas]
Devtools like vite or create-react-app have just simple solution to serve the dev server over https. This is still unsecure I know, but it is possible to create simple local certs to make https secure possible.

I couldn't find an easy solution to just start a https server with nextjs, I found out that other devtools do that a lot easier.

In vite you can start a https server like this:
```sh
vite --https
```

In create react app like this:
```sh
HTTPS=true react-scripts start
```

### Why do I need https for local development
I was building app with nextjs that included the MediaDevices api, but you can only use the api if you're running on https. Ended up using ngrok, what was ok for that project.
And I think there are a lot more reasons why running a devserver over https is usefull.

But why is their no out of the box https solution?

**Comment by uehlbran** [comment]
Vercel? Are you there? It's the people that use your software. Stop ignoring us and add this basic feature that has been requested over and over.

**Comment by MadL1me** [comment]
+1 for the feature. Its just painful not to have it

**Comment by audriuiv** [comment]
Why Vercel ignores?

**Comment by kevin-at-ogilvy** [comment]
Is there a solution for this one?

**Comment by markolazarevic9** [comment]
Is there any official way to submit request for this feature?

**Comment by kudorgyozo** [comment]
Wow, just wow. So it was a bad idea to try to learn next.js after all.

**Comment by CalebBarnes** [comment]
Since `next@13.5.1` (Sep 19, 2023) you can use `next dev --experimental-https` to use https for local development. 🙈

**Reply by duzenko** [reply]
It's supposed to add a certificate for localhost? First, it's a hack instead of a solution, second it does not work with domain names, thirdly it just fails with generic error

**Reply by CopyJosh** [reply]
It does automatically add a certificate for localhost by default, but if you pass a `-H` hostname it should sign the certificate to that domain, ie. `next dev -H somedomain.local --experimental-https`
